presto-main/src/main/java/com/facebook/presto/operator/exchange/LocalExchangeSource.java

```diff
@@ -102,14 +103,22 @@ void addPage(PageReference pageReference)
 
     public WorkProcessor<Page> pages()
     {
-        return WorkProcessor.create(() -> {
```

Comment:
Wasn't it shorter and involved less named entities previously? The method is quite compact too.
Explicit class is only needed when process/transformation keeps state.

Reply:
i can drop `Convert some processor related lambdas to classes` commit, if you wish.

Reply:
let's drop this change